Dispownership.Test/DisposableTest.cs

```diff
@@ -67,7 +64,25 @@ public void ThrowsWhenMovingABorrowedValue()
         Assert.Throws<InvalidOperationException>(() => disposable.Take());
     }
 
-    private sealed class DisposableStub : IDisposable
+    [Theory]
+    [MemberData(nameof(Disposables))]
+    public void ThrowsWhenDisposingTwice(Disposable<DisposableStub> disposable)
+    {
+#pragma warning disable IDISP007
+        disposable.Dispose();
+#pragma warning restore IDISP007
+        Assert.Throws<ObjectDisposedException>(disposable.Dispose);
+    }
+
+#pragma warning disable IDISP004
+    public static TheoryData<Disposable<DisposableStub>> Disposables()
+        => [
+            Disposable.Owned(new DisposableStub()),
+            Disposable.Borrowed(new DisposableStub()),
```

Comment:
I don't see why this test case should throw on double dispose, isn't this the point, that it is not disposed when borrowed? Maybe I just don't understand.

Reply:
Yes, the inner disposable is not disposed when borrowed. What I changed is that the wrapper tracks that it got disposed and throws in that case. I thought it would be a good thing for consistency since that's what other disposables do too.

Reply:
Oh, never mind I misunderstood the [dispose pattern](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern) 😆 

Dispose should do nothing when called multiple times, but anything that accesses the resource should throw.

Reply:
I'm not sure we should implement this for our consumers. I know this is correct according the dispose pattern as per Microsoft, but if we track dispose, the contained value can't decide if it throws when used after dispose.

This is a huge change and not necessarily in the interest of consumers, I fear? What do you think? There might be existing code that doesn't care if dispose is called, and then take after. This would be a breaking change for them, at least theoretically.

Reply:
I agree it might be a bit of a change, but I think it's a good one :)

Consumers of the disposable wrapper currently see two different behaviours when the wrapper is disposed:
* if owned, they might get an `ObjectDisposedException` when trying to do anything on the inner object
* if borrowed, they don't

One of the use cases of the disposable is to treat owned and borrowed objects the same way. Always throwing if disposed might help catch bugs :)